### PR TITLE
MAME: packaging fix

### PR DESCRIPTION
**Pull request by extrowerk**

### CODE REVIEW

File: games-emulation/mame/mame-0.209~git.recipe [hunk 1]
```diff
@@ -100,28 +107,104 @@ BUILD_PREREQUIRES="
...
+	cmd:strip$secondaryArchSuffix
 "
 
 BUILD()
 {
 	LIBS="-lbsd -lnetwork -lpugixml -lsqlite3" \
-	LDFLAGS="-lbsd -lnetwork -lpugixml -lsqlite3"  \
-	make  \
+	LDFLAGS="-lbsd -lnetwork -lpugixml -lsqlite3"	\
+	make	\
```
Comment (Begasus) [inline games-emulation/mame/mame-0.209~git.recipe]: Trailing whitespace here ;)

Comment (extrowerk) [reply games-emulation/mame/mame-0.209~git.recipe]: Thx

Comment (extrowerk) [reply games-emulation/mame/mame-0.209~git.recipe]: Thx

**Comment by extrowerk** [comment]
This PR needs https://github.com/haikuports/haikuports/pull/3915 merged.

File: games-emulation/mame/mame-0.210.recipe [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+	"
+
+ARCHITECTURES="!x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+bits=64
+commandBinDir=$binDir
+commandSuffix=$secondaryArchSuffix
+if [ "$targetArchitecture" = x86_gcc2 ]
+then
```
Comment (alaviss) [inline games-emulation/mame/mame-0.210.recipe]:
```diff
-if [ "$targetArchitecture" = x86_gcc2 ]
-then
+if [ "$targetArchitecture" = x86_gcc2 ]; then
```

File: games-emulation/mame/mame-0.210.recipe [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+prove that you have recreated the hardware faithfully?). Over time, MAME \
+(originally stood for Multiple Arcade Machine Emulator) absorbed the \
+sister-project MESS (Multi Emulator Super System), so MAME now documents a \
+wide variety of (mostly vintage) computers, video game consoles and \
+calculators, in addition to the arcade video games that were its initial focus."
+HOMEPAGE="https://www.mamedev.org/"
+COPYRIGHT="1997-2019 MAMEDev and contributors"
+LICENSE="GNU GPL v2"
+REVISION="1"
+SOURCE_URI="https://github.com/mamedev/mame/archive/mame0210.tar.gz"
```
Comment (alaviss) [inline games-emulation/mame/mame-0.210.recipe]:
```suggestion
SOURCE_URI="https://github.com/mamedev/mame/archive/mame${portVersion/./}.tar.gz"
```

File: games-emulation/mame/mame-0.210.recipe [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+CHECKSUM_SHA256="4d0b23f3c9f9b2f6283ed78d5afc9a6391d2ee6ce5f0f4bcc2a357ec14aa06c4"
+SOURCE_DIR="mame-mame0210"
```
Comment (alaviss) [inline games-emulation/mame/mame-0.210.recipe]:
```suggestion
SOURCE_FILENAME="mame-$portVersion.tar.gz"
SOURCE_DIR="mame-mame${portVersion/./}"
```

File: games-emulation/mame/mame-0.210.recipe [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+	mame$bits nltool nlwav pngcmp regrep romcmp split src2html srcclean \
+	testkeys unidasm; do
+		install -m755 $_i -t "$commandBinDir"
+		strip "$commandBinDir"/$_i
+	done
+	ln -s "$commandBinDir/mame$bits" "$commandBinDir/mame"
+
+	# Create the startscript
+	local SETTINGS_FOLDER="`finddir B_USER_SETTINGS_DIRECTORY`/mame"
+	local VENDOR_FOLDER="`finddir B_SYSTEM_LIB_DIRECTORY`/mame"
```
Comment (alaviss) [inline games-emulation/mame/mame-0.210.recipe]: As these folders can change, I think you could just add this to the script directly.

Comment (alaviss) [reply games-emulation/mame/mame-0.210.recipe]: `findpaths` can be used instead to generate a list of directories automatically.

File: games-emulation/mame/mame-0.210.recipe [hunk 6]
```diff
@@ -0,0 +1,224 @@
...
+		USE_SYSTEM_LIB_GLM=1 \
+		USE_SYSTEM_LIB_RAPIDJSON=1 \
+		USE_SYSTEM_LIB_PUGIXML=1 \
+		$jobArgs
+}
+
+INSTALL()
+{
+	# Based on https://projects.archlinux.de/svntogit/community.git/tree/trunk/PKGBUILD?h=packages/mame
+	# If you change any path here, you will have to adjust it in mame.sh.in too!
```
Comment (alaviss) [inline games-emulation/mame/mame-0.210.recipe]: No longer true.

File: games-emulation/mame/mame-0.210.recipe [hunk 7]
```diff
@@ -0,0 +1,224 @@
...
+
+	# Arch cleanup
+	# 32 bit main binary called as "mame" which would collide with our symlink
+	if [ "$targetArchitecture" = x86_gcc2 ]
+	then
```
Comment (alaviss) [inline games-emulation/mame/mame-0.210.recipe]:
```diff
-	if [ "$targetArchitecture" = x86_gcc2 ]
-	then
+	if [ "$targetArchitecture" = x86_gcc2 ]; then

File: games-emulation/mame/mame-0.210.recipe [hunk 8]
```diff
@@ -0,0 +1,224 @@
...
+
+	# Install the binaries
+	mkdir -p "$commandBinDir"
+	for _i in castool chdman floptool imgtool jedutil ldresample ldverify \
+	mame$bits nltool nlwav pngcmp regrep romcmp split src2html srcclean \
+	testkeys unidasm; do
+		install -m755 $_i -t "$commandBinDir"
+		strip "$commandBinDir"/$_i
+	done
+	ln -s "$commandBinDir/mame$bits" "$commandBinDir/mame"
```
Comment (alaviss) [inline games-emulation/mame/mame-0.210.recipe]: Why is this necessary? Does `mame` produces `mame64` and `mame32` on both arch?

If not, I think we'll be better off only having a `mame` executable.

**Comment by extrowerk** [comment]
I think it is ready

**Comment by extrowerk** [comment]
This should not harm.
